File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/domain/BaseViewModel.java
package com.kalashnyk.denys.gmapsapp.domain;


import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;

import com.kalashnyk.denys.App;

abstract class BaseViewModel extends AndroidViewModel {

    public BaseViewModel(Application application) {
        super(application);
    }

    public App getContext() {
        return getApplication();
    }
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/domain/BaseViewModel.kt
package com.kalashnyk.denys.kotlinbindingsample.domain

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.support.annotation.StringRes
import com.kalashnyk.denys.kotlinbindingsample.App

abstract class BaseViewModel(application: Application) : AndroidViewModel(application) {
    fun getContext() = getApplication<App>()
    fun getString(@StringRes id: Int): String = getContext().getString(id)
}